### ISSUE TITLE
Apply pipeTransport.quoteArgs to pickRemoteProcess command

### ISSUE THREAD

**Issue by daandemeyer** [microsoft/vscode-cpptools#6702]
Type: Debugger

**Describe the bug**
- OS and Version: Arch Linux
- VS Code Version: 1.52.1
- C/C++ Extension Version: 1.1.3
- Other extensions you installed (and if the issue persists after disabling them): Not relevant

When running the `pickRemoteProcess` command via `pipeTransport`, `quoteArgs` is not applied to `pipeArgs` and the args are quoted regardless of the value of `quoteArgs`.

**To Reproduce**

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "cppdbg",
            "program": "/usr/lib/systemd/systemd-resolved",
            "processId": "${command:pickRemoteProcess}",
            "request": "attach",
            "name": "Attach",
            "pipeTransport": {
                "pipeCwd": "${workspaceFolder}",
                "pipeProgram": "mkosi",
                "pipeArgs": [
                    "ssh"
                ],
                "debuggerPath": "/usr/bin/gdb",
                "quoteArgs": false
            }
        }
    ]
}
```

Produces in the error log:

`Command failed: "mkosi" "ssh" 'sh -c "uname && if [ $(uname) = \"Linux\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= ; elif [ $(uname) = \"Darwin\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= -c; fi"'`

We can see "ssh" is quoted even though quoteArgs is set to false.

**Comment by sean-mcmanus** [comment]
FYI, @WardenGnaw might be OOF till January.

**Comment by WardenGnaw** [comment]
@DaanDeMeyer  Are you expecting the command to be:
```
"mkosi" ssh 'sh -c "uname && if [ $(uname) = \"Linux\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= ; elif [ $(uname) = \"Darwin\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= -c; fi"'
```

At the moment, we only use quoteArgs for the debug adapter. 

This would require a change in this [code](https://github.com/microsoft/vscode-cpptools/blob/2ad35c83c0df5462aa85b350fea52fc1f23ef07d/Extension/src/Debugger/attachToProcess.ts#L89) path to check for quoteArgs.

**Comment by DaanDeMeyer** [comment]
Yes, that's what I was expecting. My error turned out to be caused by something else but I figure this might be useful to have nonetheless.
